Parallel filesystem creation

aughr opened the pull request:
Fixes #681. It caps the concurrency at GOMAXPROCS.

There are a couple smaller changes in here too, one necessary and one useful:
* I added a flag to allow passing a pattern specifying which blackbox tests will be run, since `test.run` only sees the overall single test method.
* I ensured that In partitions always have a Device and MountPath set, even if they're blank. That way you can take a blank partition and format it.

lucab commented:
ok to test

lucab commented on internal/exec/stages/disks/filesystems.go:
```diff
@@ -57,12 +58,36 @@ func (s stage) createFilesystems(config types.Config) error {
+
+	// Return combined errors
+	var errs []string
+	for _, _ = range fss {
```
As per travis failure, this should be:
```
for range fss {
```

lucab commented on tests/filesystem.go:
```diff
@@ -253,7 +251,10 @@ func setupDisk(ctx context.Context, disk *types.Disk, diskIndex int, imageSize i
 		}()
 
-		partition.Device = fmt.Sprintf("%sp%d", disk.Device, partition.Number)
+		if partition.TypeCode == "blank" || partition.FilesystemType == "" || partition.FilesystemType == "swap" {
```
What's the reason for moving this here, after the `Mkdir`?

ajeddeloh replied:
This might have caused the BB test failures.

aughr replied:
Was hoping this would just work…

The goal is to make sure that blank partitions end up with a Device and MountPath set, so that if it's later formatted, the tests know what to inspect.

I'll take a look at the failures.

ajeddeloh commented on tests/blackbox_test.go:
```diff
@@ -43,6 +45,8 @@ var (
 	// somewhat of an abuse of contexts but go's got our hands tied
 	killContext = context.TODO()
+
+	runPattern = flag.String("blackbox.run", "", "run only blackbox tests matching regexp")
```
This can already be done natively with the golang test runner. You can specify a particular test with something like ` -test.run  TestIgnitionBlackBox/Create_a_Directory_on_the_Root_Filesystem_2.3.0`. This commit can be dropped. 

cc @arithx since he was curious if this was possible.

arithx replied:
@ajeddeloh do patterns work though? Because this will allow for regexp pattern matching against subtest names as well.

ajeddeloh replied:
Yeah, or at least globbing does. I can throw a * at the end to get all the versions.

ajeddeloh replied:
Looks like this got squashed into the test commit.

aughr replied:
Ugh, thank you.

aughr commented:
@ajeddeloh Unable to log into the Black Box Jenkins… "Failed to return user's identity." after Google sign-in, even in a private browser session.

ajeddeloh commented:
Arg they're on a private jenkins. Here's the log:
https://gist.github.com/ajeddeloh/6080c0d343b31951522bcfaf75454de9

aughr commented:
OK, feedback addressed. The changes to the test support code for blank filesystems are now more tightly scoped, only preventing trying to create files on a blank FS. I've also rewritten the commit messages and removed the `blackbox.run` flag I added.

aughr commented:
The blackbox.run flag is really gone now.

ajeddeloh commented:
Thanks for the PR!
